api/v3/Payment.php

```diff
@@ -187,6 +197,119 @@ function _civicrm_api3_payment_create_spec(&$params) {
+      'bao' => 'CRM_Financial_DAO_FinancialTrxn',
+      'localizable' => 0,
+      'html' => [
+        'type' => 'Select',
+      ],
+      'pseudoconstant' => [
+        'optionGroupName' => 'accept_creditcard',
+        'optionEditPath' => 'civicrm/admin/options/accept_creditcard',
+      ],
+    ],
+    'trxn_result_code' => [
+      'name' => 'trxn_result_code',
+      'type' => CRM_Utils_Type::T_STRING,
+      'title' => ts('Transaction Result Code'),
+      'description' => ts('processor result code'),
+      'maxlength' => 255,
+      'size' => CRM_Utils_Type::HUGE,
+      'where' => 'civicrm_financial_trxn.trxn_result_code',
+      'table_name' => 'civicrm_financial_trxn',
+      'entity' => 'FinancialTrxn',
+      'bao' => 'CRM_Financial_DAO_FinancialTrxn',
+      'localizable' => 0,
+    ],
+    'trxn_id' => [
+      'name' => 'trxn_id',
+      'type' => CRM_Utils_Type::T_STRING,
+      'title' => ts('Transaction ID'),
+      'description' => ts('Transaction id supplied by external processor. This may not be unique.'),
+      'maxlength' => 255,
+      'size' => 10,
+      'where' => 'civicrm_financial_trxn.trxn_id',
+      'table_name' => 'civicrm_financial_trxn',
+      'entity' => 'FinancialTrxn',
+      'bao' => 'CRM_Financial_DAO_FinancialTrxn',
+      'localizable' => 0,
+      'html' => [
+        'type' => 'Text',
+      ],
+    ],
+    'check_number' => [
+      'name' => 'check_number',
+      'type' => CRM_Utils_Type::T_STRING,
+      'title' => ts('Check Number'),
+      'description' => ts('Check number'),
+      'maxlength' => 255,
+      'size' => 6,
+      'where' => 'civicrm_financial_trxn.check_number',
+      'table_name' => 'civicrm_financial_trxn',
+      'entity' => 'FinancialTrxn',
+      'bao' => 'CRM_Financial_DAO_FinancialTrxn',
+      'localizable' => 0,
+      'html' => [
+        'type' => 'Text',
+      ],
+    ],
+    'pan_truncation' => [
+      'name' => 'pan_truncation',
+      'type' => CRM_Utils_Type::T_STRING,
+      'title' => ts('Pan Truncation'),
+      'description' => ts('Last 4 digits of credit card'),
+      'maxlength' => 4,
+      'size' => 4,
+      'where' => 'civicrm_financial_trxn.pan_truncation',
+      'table_name' => 'civicrm_financial_trxn',
+      'entity' => 'FinancialTrxn',
+      'bao' => 'CRM_Financial_DAO_FinancialTrxn',
+      'localizable' => 0,
+      'html' => [
+        'type' => 'Text',
+      ],
+    ],
+    'order_reference' => [
+      'name' => 'order_reference',
+      'type' => CRM_Utils_Type::T_STRING,
+      'title' => ts('Order Reference'),
+      'description' => ts('Payment Processor external order reference'),
+      'maxlength' => 255,
+      'size' => 25,
+      'where' => 'civicrm_financial_trxn.order_reference',
+      'table_name' => 'civicrm_financial_trxn',
+      'entity' => 'FinancialTrxn',
+      'bao' => 'CRM_Financial_DAO_FinancialTrxn',
+      'localizable' => 0,
+      'html' => [
+        'type' => 'Text',
+      ],
+    ],
```

Comment:
@eileenmcnaughton `order_reference` was only added in 5.20 - will this cause problems on 5.19 as the column won't exist in the DB?

Reply:
Yeah - I think it's the cause of the test fails - I just pulled it out again